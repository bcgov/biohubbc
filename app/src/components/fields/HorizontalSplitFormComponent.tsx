import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import React from 'react';

export interface IHorizontalSplitFormComponentProps {
  title: string;
  summary: string;
  component: any;
}

/**
 * Shared component for various survey sections
 *
 * @return {*}
 */
const HorizontalSplitFormComponent: React.FC<IHorizontalSplitFormComponentProps> = (props) => {
  const { title, summary, component } = props;

  return (
    <>
      <Box width="100%" display="flex" flexWrap="nowrap">
        <Box width="30%" p={5}>
          <Typography variant="h2">{title}</Typography>
          <Box pt={3}>
            <Typography variant="body2">{summary}</Typography>
          </Box>
        </Box>
        <Box width="70%" p={5}>
          {component}
        </Box>
      </Box>
    </>
  );
};

export default HorizontalSplitFormComponent;
