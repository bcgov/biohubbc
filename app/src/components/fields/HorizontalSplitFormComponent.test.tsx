import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { render } from '@testing-library/react';
import React from 'react';
import HorizontalSplitFormComponent from './HorizontalSplitFormComponent';

describe('HorizontalSplitFormComponent', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <HorizontalSplitFormComponent
        title="Section title"
        summary="Section summary"
        component={
          <Box>
            <Typography>Hey this is my content</Typography>
          </Box>
        }
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
