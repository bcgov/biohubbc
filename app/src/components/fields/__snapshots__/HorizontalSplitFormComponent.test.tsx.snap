// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalSplitFormComponent renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2"
      >
        Section title
      </h2>
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Section summary
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Hey this is my content
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
